storage/wal/wal_bench_test.go

```diff
@@ -0,0 +1,76 @@
+package wal_test
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/nvanbenschoten/rafttoy/storage/engine"
+	"github.com/nvanbenschoten/rafttoy/storage/wal"
+	"github.com/nvanbenschoten/rafttoy/util"
+	"github.com/nvanbenschoten/rafttoy/workload"
+	"go.etcd.io/etcd/raft/raftpb"
+)
+
+func benchmarkWalConfig(b *testing.B, batchSize, bytes int, w wal.Wal) {
+	workers := workload.NewWorkers(workload.Config{
+		KeyPrefix: engine.MinDataKey,
+		KeyLen:    len(engine.MinDataKey) + 8,
+		ValLen:    bytes,
+		Workers:   1,
+		Proposals: b.N,
+	})
+	worker := workers[0]
+
+	entries := make([]raftpb.Entry, 0, batchSize)
+	appendEntries := func() {
+		if len(entries) > 0 {
+			w.Append(entries)
+			entries = entries[:0]
+		}
+	}
+
+	b.ResetTimer()
+	var index uint64
+	for prop := worker.NextProposal(); prop != nil; prop = worker.NextProposal() {
+		index++
+		e := raftpb.Entry{
+			Index: index,
+			Type:  raftpb.EntryNormal,
+			Data:  append([]byte(nil), prop...),
```

Comment:
Should we try to recycle memory here?

Reply:
I thought about copying in bufalloc.ByteAllocator, but figured it was overkill. Done. allocs/op on the etc benchmarks all went down by 1. pebble looks more variable